common/common-rest/src/main/java/org/apache/servicecomb/common/rest/AbstractRestInvocation.java

```diff
@@ -249,4 +265,21 @@ protected void onExecuteHttpServerFiltersFinish(Response response, Throwable e)
       invocation.onFinish(response);
     }
   }
+
+  public void setAfterCreateInvocationHandler(
```

Comment:
Maybe create a new protected method is better.
protected void afterCreateInvocation()
protected void beforeCreateInvocation()
protected void createInvocation()

 And we can implement it in RestProducerInvocation, e.g
RestProducerInvocation invo = new RestProducerInvocation.java(RoutingContext context)
..

Reply:
I guess you mean to replace the `AfterCreateInvocationHandler` with the method `protected void afterCreateInvocation()` ?
But currently we cannot get RoutingContext directly in `AbstractRestInvocation`, unless we cast the type of  `AbstractRestInvocation.requestEx` and add a getter into `VertxServerRequestToHttpServletRequest` like below:
```java
protected void afterCreateInvocation() {
  if (!VertxServerRequestToHttpServletRequest.class.isAssignableFrom(this.requestEx.getClass())) {
    return;
  }
  VertxServerRequestToHttpServletRequest request = (VertxServerRequestToHttpServletRequest) this.requestEx;
  RoutingContext routingContext = request.getContext();
  routingContext.put(RestConst.REST_INVOCATION_CONTEXT, this.invocation);
}
```

Am I right?

Reply:
The code has been refactored, please review the new commit